rbd/snapshot_nautilus.go

```diff
@@ -0,0 +1,68 @@
+// +build !luminous,!mimic
+//
+// Ceph Nautilus introduced rbd_get_parent() and deprecated rbd_get_parent_info().
+// Ceph Nautilus introduced rbd_list_children3() and deprecated rbd_list_children().
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <rbd/librbd.h>
+// #include <errno.h>
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+)
+
+// int rbd_get_parent(rbd_image_t image, rbd_linked_image_spec_t *parent_image,
+//                    rbd_snap_spec_t *parent_snap)
+func (image *Image) GetParentInfo(p_pool, p_name, p_snapname []byte) error {
+	if err := image.validate(imageIsOpen); err != nil {
+		return err
+	}
+
+	parent_image := C.rbd_linked_image_spec_t{}
+	parent_snap := C.rbd_snap_spec_t{}
+	ret := C.rbd_get_parent(image.image, &parent_image, &parent_snap)
+	if ret != 0 {
+		return RBDError(ret)
+	}
+
+	defer C.rbd_linked_image_spec_cleanup(&parent_image)
+	defer C.rbd_snap_spec_cleanup(&parent_snap)
+
+	strlen := int(C.strlen(parent_image.pool_name))
+	if len(p_pool) < strlen {
+		return fmt.Errorf("can not fix %d bytes in parent pool of %d bytes", strlen, len(p_pool))
+	}
+
+	go_p_pool := []byte(C.GoString(parent_image.pool_name))
+	for i := 0; i < strlen; i++ {
```

Comment:
Use the go copy() builtin here.

Reply:
I wonder why I put `//copy(...)` in comments just below... Testing now and things seem to work fine. Will correct this with the next update of this PR.